actors/multisig/tests/multisig_actor_test.rs

```diff
@@ -1847,23 +1847,112 @@ fn test_lock_balance_checks_preconditions() {
+    use super::*;
 
-    let mut rt = construct_runtime(msig);
-    let h = util::ActorHarness::new();
-    let signers = vec![anne, bob, chuck];
-    let initial_threshold = 2;
+    #[test]
+    fn test_change_threshold() {
+        let msig = Address::new_id(100);
+        let anne = Address::new_id(101);
+        let bob = Address::new_id(102);
+        let chuck = Address::new_id(103);
 
-    h.construct_and_verify(&mut rt, initial_threshold, 0, 0, signers);
+        struct TestCase<'a> {
+            #[allow(dead_code)]
+            desc: &'a str,
+            initial_threshold: u64,
+            set_threshold: u64,
+            code: ExitCode,
+        }
 
-    rt.set_caller(*MULTISIG_ACTOR_CODE_ID, msig);
-    let ret = h.change_num_approvals_threshold(&mut rt, 1).unwrap();
-    assert_eq!(RawBytes::default(), ret);
-    let st: State = rt.get_state();
-    assert_eq!(1, st.num_approvals_threshold);
+        let test_cases = vec![
+            TestCase {
+                desc: "happy path decrease threshold",
+                initial_threshold: 2,
+                set_threshold: 1,
+                code: ExitCode::OK,
+            },
+            TestCase {
+                desc: "happy path simple increase threshold",
+                initial_threshold: 2,
+                set_threshold: 3,
+                code: ExitCode::OK,
+            },
+            TestCase {
+                desc: "fail to set threshold to zero",
+                initial_threshold: 2,
+                set_threshold: 0,
+                code: ExitCode::USR_ILLEGAL_ARGUMENT,
+            },
+            TestCase {
+                desc: "fail to set threshold above number of signers",
+                initial_threshold: 2,
+                set_threshold: 4,
+                code: ExitCode::USR_ILLEGAL_ARGUMENT,
+            },
+        ];
+
+        for tc in test_cases {
+            let mut rt = construct_runtime(msig);
+            let h = util::ActorHarness::new();
+            let signers = vec![anne, bob, chuck];
+            h.construct_and_verify(&mut rt, tc.initial_threshold, 0, 0, signers);
+            rt.set_caller(*MULTISIG_ACTOR_CODE_ID, msig);
+            let ret = h.change_num_approvals_threshold(&mut rt, tc.set_threshold);
+            match tc.code {
+                ExitCode::OK => {
+                    assert_eq!(RawBytes::default(), ret.unwrap());
+                    let st: State = rt.get_state();
+                    assert_eq!(tc.set_threshold, st.num_approvals_threshold);
+                }
+                _ => {
+                    assert_eq!(
+                        tc.code,
+                        ret.expect_err("change threshold return expected to be actor error")
+                            .exit_code()
+                    );
+                }
+            }
+        }
+    }
+
+    #[test]
+    fn tx_can_be_reapproved_and_execed_after_threshold_lowered() {
+        let msig = Address::new_id(100);
```

Comment:
I guess we could extract those addresses and `signers` as constants?

Reply:
Agreed. I've tried it both ways and feel pretty ambivalent so I'm just going to keep it like this for now.